shared/actions/fs.js

```diff
@@ -96,14 +96,18 @@ function* download(action: FsGen.DownloadPayload): Saga.SagaGenerator<any, any>
     },
   })
 
-  // Fake out progress until we have the real thing.
-  // TODO: have the real thing.
-  const total = 6
-  for (let progress = 0; progress < total - 1; ++progress) {
+  let progress
+  do {
     yield Saga.delay(500)
-    yield Saga.call(RPCTypes.SimpleFSSimpleFSCheckRpcPromise, {opID})
-    yield Saga.put(FsGen.createFileTransferProgress({key, completePortion: progress / total}))
-  }
+    progress = yield Saga.call(RPCTypes.SimpleFSSimpleFSCheckRpcPromise, {opID})
```

Comment:
what happens when there are problems? like an error or etc

Reply:
Thanks for bringing it up! I haven't dealt with any error case at all, but have a ticket to handle cancels. I'll add into that ticket to handle RPC errors as well.